tests/cpp_unit_tests/test_observability.cpp

```diff
@@ -86,11 +89,21 @@ TEST_CASE("Necessary observability check") {
 
         SUBCASE("Parallel branch get counted as one sensor") {
             // Add sensor on branch 3 to side. Hence 2 parallel sensors
+            topo.power_sensors_per_branch_from = {from_sparse, {0, 0, 1, 1}};
             topo.power_sensors_per_branch_to = {from_sparse, {0, 0, 0, 1}};
             se_input.measured_branch_to_power = {{1.0, 1.0, 1.0}};
             check_not_observable(topo, param, se_input);
         }
     }
+    SUBCASE("Not independent") {
+        // set branch sensor to bus_1 <-branch_1-> bus_2
+        // it is not independent with injection sensor of bus_2
+        topo.power_sensors_per_branch_from = {from_sparse, {0, 0, 1, 1}};
+        // set non phasor measurement
+        se_input.measured_voltage = {{{1.0, nan}, 1.0}};
+        // this will throw NotObservableError
+        check_not_observable(topo, param, se_input);
+    }
```

Comment:
can we add another case with a meshed grid or is that overkill?

Reply:
I don't think it is necessary here.